static/sw.js

```diff
@@ -16,34 +14,53 @@ self.addEventListener("install", function (event) {
 self.addEventListener("fetch", function (event) {
   const requestUrl = new URL(event.request.url);
 
+  // Ignore non-HTTP/HTTPS requests
+  if (!requestUrl.protocol.startsWith("http")) {
+    return;
+  }
+
+  // Bypass cache for external image requests
+  if (
+    requestUrl.origin !== location.origin &&
+    /\.(jpe?g|png|webp)$/.test(requestUrl.pathname)
+  ) {
+    event.respondWith(fetch(event.request));
+    return;
+  }
+
   event.respondWith(
-    // Try to fetch from network first
-    fetch(event.request)
-      .then(function (networkResponse) {
-        // If request is successful, clone and cache the response
-        let cacheName = requestUrl.pathname;
-
-        // Choose the cache depending on the file type
-        if (/\.woff2?$|\.ttf$/.test(requestUrl.pathname)) {
-          cacheName = FONT_CACHE_CYF;
-        } else if (/\.jpe?g$|\.png$|\.webp$/.test(requestUrl.pathname)) {
-          cacheName = IMAGE_CACHE_CYF;
-        } else {
-          cacheName = CACHE_CYF;
-        }
-
-        caches.open(cacheName).then(function (cache) {
-          cache.put(event.request, networkResponse.clone());
-        });
+    caches.match(event.request).then(function (cachedResponse) {
+      var fetchPromise = fetch(event.request)
+        .then(function (networkResponse) {
+          if (!networkResponse || networkResponse.status !== 200) {
+            return cachedResponse;
+          }
+
+          let cacheName = CACHE_CYF;
+          let responseToCache = networkResponse.clone();
 
-        // Return the network response
-        return networkResponse;
-      })
-      .catch(function () {
-        // If network request fails, try to get from cache
-        return caches.match(event.request).then(function (cacheResponse) {
-          return cacheResponse;
+          // Choose the cache depending on the file type
+          if (/\.woff2?$|\.ttf$/.test(requestUrl.pathname)) {
+            cacheName = FONT_CACHE_CYF;
```

Comment:
It appears that font caching isn't working both in the Netlify preview and locally for me.

For some reason in the Netlify preview, it appears the font cache is never initiated:

<img width="351" alt="Screenshot 2023-07-01 at 11 37 07" src="https://github.com/CodeYourFuture/curriculum/assets/424411/6c408855-bed0-46ae-af12-1afaddc5cad2">

Locally, I do see a font cache, but it's empty:

<img width="1282" alt="Screenshot 2023-07-01 at 11 39 03" src="https://github.com/CodeYourFuture/curriculum/assets/424411/1d6f31fb-0211-4662-991b-398acfb70b6b">

It's not entirely clear to me why this is, but I think it's related to us loading via Google Fonts - since GF loads the actual font files as a second step (i.e. `https://fonts.googleapis.com/css2?family=Space+Grotesk:wght@300;500;700&family=Inter:wght@400;600&display=swap` will then request `https://fonts.gstatic.com/s/spacegrotesk/v15/V8mDoQDjQSkFtoMM3T6r8E7mPbF4C_k3HqU.woff2`)